### DISCUSSION TITLE
Smart breaker and  Riemann sum integral sensor not working properly

### DISCUSSION THREAD

**Discussion by dankony** [make-all/tuya-local#1192] [Q&A] [answered]
Hi all! I've bought this smart circuit breaker https://www.aliexpress.com/item/1005004153509348.html. I use it to switch on/off a water heating boiler, so I'm using the circuit breaker itself for that. Power monitor in HA (thanks to Tuya local) and in Smart life app are correct, however when I use Riemann sum integral sensor to show Energy usage I get the following issue: After the breaker has been turned on the value of Riemann sum integral jumps straight away (for example - from 581204 Wh to 604070 Wh) so the measurment becomes wrong... If the boiler itself switches off (the boiler's internal breaker) and after some time ON again, the value of Riemann sum integral is what it shoud be, and the measurement is ok.
The problem is only when the breaker is being turned off for some time and back on again (however it is still connected to power, wi-fi and HA)
Any ideas?
Thank you!

P.S.
I have 3 other different smart sockets and switches - they don't show that behaviour. Latest firmware everywhere.

**Comment by make-all** [comment, marked as answer]
It sounds like the device does not update the power sensor when the switch is off, so Home Assistant doesn't see it go to zero, and instead estimates the energy use for the off period based on the last power reading.  Does the device have a specific config, or is it using one of the generic smartswitch configs?  We could force the power to 0 when the switch is off, though it isn't really correct as the wifi on the switch itself usually uses around 1W when the switch is off, and some smartplugs show that (others are calibrated to hide it, which is effectively what this solution does)

**Reply by dankony** [reply]
If I understood the question rigth, the config is generic - smartplugv2_energyv3. 
And probbably you are right about the "estimation" :) A 1W error is better than 2kW error, so... how do we force the power to go to 0?

**Reply by make-all** [reply]
I have made an update to that config to conditionally force power to 0 when the switch is off.

**Reply by dankony** [reply]
Thank you very much for that!
I will try and report back!

**Reply by dankony** [reply]
It works as it should! Thanks, you got a coffee!
